rpm/0003-Move-static-libraries-to-separate-cmakelists-files-a.patch

```diff
@@ -0,0 +1,202 @@
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Some static cmake libraries where only present inside the adb
+CMakeLists which did not seem right and makes builds without adb fail.
+
+Signed-off-by: Björn Bidar <bjorn.bidar@jolla.com>
+---
+ vendor/CMakeLists.adb.txt            | 52 ----------------------------
+ vendor/CMakeLists.fastboot.txt       |  1 +
+ vendor/CMakeLists.libbase.txt        |  2 +-
+ vendor/CMakeLists.libcutils.txt      | 34 ++++++++++++++++++
+ vendor/CMakeLists.libdiagnoseusb.txt |  7 ++++
+ vendor/CMakeLists.liblog.txt         | 16 +++++++++
+ vendor/CMakeLists.txt                |  3 ++
+ 7 files changed, 62 insertions(+), 53 deletions(-)
+ create mode 100644 vendor/CMakeLists.libcutils.txt
+ create mode 100644 vendor/CMakeLists.libdiagnoseusb.txt
+ create mode 100644 vendor/CMakeLists.liblog.txt
+
+diff --git a/vendor/CMakeLists.adb.txt b/vendor/CMakeLists.adb.txt
+index 7f59783461252a49ba44008842d8f8b5f60e982e..c135ac59f54861946cf2dee28bc0233098a3ba4a 100644
+--- a/vendor/CMakeLists.adb.txt
++++ b/vendor/CMakeLists.adb.txt
+@@ -90,58 +90,6 @@ target_include_directories(libadb_tls_connection_defaults PUBLIC
+ 	boringssl/include
+ 	libbase/include)
+ 
+-add_library(liblog STATIC
+-	logging/liblog/log_event_list.cpp
+-	logging/liblog/log_event_write.cpp
+-	logging/liblog/logger_name.cpp
+-	logging/liblog/logger_read.cpp
+-	logging/liblog/logger_write.cpp
+-	logging/liblog/properties.cpp
+-	logging/liblog/logprint.cpp)
+-
+-target_compile_definitions(liblog PRIVATE
+-	-DLIBLOG_LOG_TAG=1006 -D_XOPEN_SOURCE=700 -DFAKE_LOG_DEVICE=1)
+-target_include_directories(liblog PUBLIC
+-	core/include
+-	logging/liblog/include
+-	core/libcutils/include
+-	libbase/include)
+-
+-set(libcutils_SOURCES
+-	core/libcutils/android_get_control_file.cpp
+-	core/libcutils/ashmem-host.cpp
+-	core/libcutils/config_utils.cpp
+-	core/libcutils/fs.cpp
+-	core/libcutils/hashmap.cpp
+-	core/libcutils/iosched_policy.cpp
+-	core/libcutils/load_file.cpp
+-	core/libcutils/multiuser.cpp
+-	core/libcutils/native_handle.cpp
+-	core/libcutils/properties.cpp
+-	core/libcutils/record_stream.cpp
+-	core/libcutils/socket_inaddr_any_server_unix.cpp
+-	core/libcutils/socket_local_client_unix.cpp
+-	core/libcutils/socket_local_server_unix.cpp
+-	core/libcutils/socket_network_client_unix.cpp
+-	core/libcutils/sockets_unix.cpp
+-	core/libcutils/sockets.cpp
+-	core/libcutils/str_parms.cpp
+-	core/libcutils/strlcpy.c
+-	core/libcutils/trace-host.cpp)
+-
+-if (NOT APPLE AND NOT WIN32)
+-	list(APPEND libcutils_SOURCES
+-		core/libcutils/canned_fs_config.cpp
+-		core/libcutils/fs_config.cpp)
+-endif()
+-
+-add_library(libcutils STATIC ${libcutils_SOURCES})
+-target_compile_definitions(libcutils PRIVATE -D_GNU_SOURCE)
+-target_include_directories(libcutils PRIVATE
+-	logging/liblog/include core/libutils/include)
+-target_include_directories(libcutils PUBLIC
+-	core/libcutils/include libbase/include)
+-
+ add_library(libdiagnoseusb STATIC
```

Comment:
Should this  libdiagnoseusb be also removed from here?

Reply:
Why should it? The library is needed to compile adb.

Reply:
Because it was split in vendor/CMakeLists.libdiagnoseusb.txt like the others, like the others.